src/main/java/com/robinpowered/sdk/model/DeviceManifest.java

```diff
@@ -52,7 +52,7 @@ public DeviceManifest(int id, String name, String slug, Boolean isRelay, String
 
 
     @Override
-    public int getId() {
```

Comment:
You can't use primitives with generics. Agree on the `null` thing, though a simple inspection will see that this is only set via the constructor which uses the primative and the internal property is a primative `int`. Push comes to shove we can include a new annotations dependency to add a `@NotNull` annotation, but that wouldn't be monitored by android studio anyway, at least as far as I know, since it's not the android-support `NonNull`. This obviously improves with Java8, which we can exclusively move to when Android N allows it.